it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionControllerISpec.scala

```diff
@@ -251,14 +263,24 @@ class SubscriptionControllerISpec extends BaseControllerISpec with SessionDataMi
         checkHtmlResultWithBodyText(result, "Add your agency information")
       }
 
-      "telephone is invalid" in {
+      "telephone is invalid with only words" in {
         AuthStub.hasNoEnrolments(subscribingAgent)
         sessionStoreService.knownFactsResult = Some(myAgencyKnownFactsResult)
 
-        val result = await(controller.submitSubscriptionDetails(subscriptionDetailsRequest("telephone", Seq("telephone" -> "12345"))))
+        val result = await(controller.submitSubscriptionDetails(subscriptionDetailsRequest("telephone", Seq("telephone" -> "nine"))))
 
         status(result) shouldBe 200
-        checkHtmlResultWithBodyText(result, "Add your agency information")
+        checkHtmlResultWithBodyText(result, "Add your agency information", "Please enter a valid telephone number")
+      }
+
+      "telephone is invalid with numbers and words" in {
```

Comment:
It would be better to just have one `"telephone is invalid"` test at the integration level (i.e. in this file) and then have tests for all the different reasons a telephone number might be invalid at the unit level (e.g. in FieldMappingsSpec). Have a look and see if this is possible.